app/src/main/java/com/asfoundation/wallet/repository/BdsBackEndWriter.kt

```diff
@@ -0,0 +1,37 @@
+package com.asfoundation.wallet.repository
+
+import com.asfoundation.wallet.interact.FindDefaultWalletInteract
+import com.asfoundation.wallet.poa.Proof
+import com.asfoundation.wallet.poa.ProofSubmissionFeeData
+import com.asfoundation.wallet.poa.ProofWriter
+import com.asfoundation.wallet.service.PoASubmissionService
+import io.reactivex.Single
+import java.math.BigDecimal
+import java.net.UnknownHostException
+
+class BdsBackEndWriter(private val defaultWalletInteract: FindDefaultWalletInteract,
+                       private val service: PoASubmissionService) : ProofWriter {
+
+  override fun writeProof(proof: Proof): Single<String> {
+    return defaultWalletInteract.find()
+        .flatMap { wallet -> service.submitProof(proof, wallet.address) }
+  }
+
+  override fun hasEnoughFunds(chainId: Int): Single<ProofSubmissionFeeData> {
+    return defaultWalletInteract.find().flatMap {
```

Comment:
consider using map instead of this flatmap

Reply:
Done